paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/source/ContinuousFileStoreSource.java

```diff
@@ -40,6 +48,7 @@ public class ContinuousFileStoreSource extends FlinkSource {
 
     protected final Map<String, String> options;
     protected final BucketMode bucketMode;
+    protected ConcurrentHashMap<Integer, Long> consumedRecords = new ConcurrentHashMap<>();
```

Comment:
`SourceReader` runs in task manager, while `SplitEnumerator` runs in job manager. They can't communicate only through memory.